cmd/thv-operator/controllers/mcpserver_controller.go

```diff
@@ -186,6 +188,15 @@ func (r *MCPServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Validate PodTemplateSpec early - before other validations
+	// This ensures we fail fast if the spec is invalid
+	if !r.validateAndUpdatePodTemplateStatus(ctx, mcpServer) {
+		// Invalid PodTemplateSpec - return without error to avoid infinite retries
+		// The user must fix the spec and the next reconciliation will retry
+		ctxLogger.Info("Skipping reconciliation due to invalid PodTemplateSpec")
```

Comment:
Could we do a `Warn` here maybe if we're worried about infinite retries from error? Something about reporting an invalid PodTemplateSpec as `Info` gives me the shivers..

Reply:
`Warn` is not available with this logger. I'll remove this log message since we anyway use the event recorder.

Reply:
Oh damn, is that a good thing, or should we replace the logger with one thats more level friendly?

Reply:
I resolved it because I ended up removing this log message. But... I don't know :smile: we're just cargoculting what k8s libraries use/expect.